test/extended/dns/dns.go

```diff
@@ -554,4 +557,61 @@ var _ = Describe("[sig-network-edge] DNS", func() {
 		pod := createDNSPod(f.Namespace.Name, cmd)
 		validateDNSResults(f, pod, expect, times)
 	})
+
+	It("should answer queries using the local DNS endpoint", func() {
+		ctx := context.Background()
+
+		By("starting an exec pod for running queries")
+
+		queryPodExec, err := exutil.NewPodExecutor(oc, "dnsquery", image.ShellImage())
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		queryPod, err := f.ClientSet.CoreV1().Pods(oc.Namespace()).Get(ctx, "dnsquery", metav1.GetOptions{})
+		if err != nil {
+			e2e.Failf("unable to get query pod that was just started: %v", err)
+		}
+		e2e.Logf("DNS query pod is on node: %q", queryPod.Spec.NodeName)
+
+		By("finding the exec pod's local DNS endpoint")
+
+		// We find the local dns pod by filtering for pods in openshift-dns namespace that are on the same node,
+		// then making sure the pod is running, not being deleted, and is owned by the default dns daemonset.
+		// Note: We expect there to be a pod co-located with query pod at all times, even if there is a taint applied
+		// to the cluster under test. The taint should taint both the query pod and DNS pod together.
+		nodeSelector := metav1.ListOptions{FieldSelector: fields.OneTermEqualSelector("spec.nodeName", queryPod.Spec.NodeName).String()}
+		dnsPodList, err := f.ClientSet.CoreV1().Pods("openshift-dns").List(ctx, nodeSelector)
+		if err != nil {
+			e2e.Failf("unable to get local DNS pod: %v", err)
+		}
+		var localDnsPod *kapiv1.Pod
+		for _, pod := range dnsPodList.Items {
+			if pod.Status.Phase == kapiv1.PodRunning && pod.DeletionTimestamp == nil {
+				for _, ownerReference := range pod.ObjectMeta.OwnerReferences {
+					if ownerReference.Name == "dns-default" {
+						localDnsPod = pod.DeepCopy()
+						break
+					}
+				}
+			}
+		}
+		if localDnsPod == nil {
+			e2e.Failf("unable to find a local DNS pod on node %q", queryPod.Spec.NodeName)
+		}
+		e2e.Logf("Local DNS Pod: %q", localDnsPod.Name)
+
+		// This queries for the hostname of the server responding to the DNS request enabled by the chaos plugin.
+		// More information: https://coredns.io/plugins/chaos/
+		hostnameDig := "dig +short +notcp +noall +answer CH TXT hostname.bind"
+		attempts := 10
+
+		By("running this command " + fmt.Sprintf("%v", attempts) + " times: " + hostnameDig + "\n")
+		for i := 1; i <= attempts; i++ {
+			digOut, err := queryPodExec.Exec(hostnameDig)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			if strings.Trim(digOut, "\"") != localDnsPod.Name {
```

Comment:
Can this be turned into a `o.Expect()` condition?

Reply:
I tried on this one:
`o.Expect(strings.Trim(digOut, "\"")).To(o.Equal(localDnsPod.Name),"expected DNS chaos query to always use the local DNS pod")`

But I think it get's a little harder to read with all that's going on. I prefer just leaving in the If statement, let me know if you disagree.